src/jl_exported_data.list

```diff
@@ -0,0 +1,113 @@
+jl_abstractarray_type
```

Comment:
Any particular reason not to make this a normal `.c` file? They're all just trivially `void *` variable decls.

Reply:
To keep symmetry with `jl_exported_funcs.list`; but it's not that big an issue.  It seemed nice to have the `src/Makefile` produce `.list` files, then have those `.list` files be processed into a `.c` file, in a symmetric fashion.

Reply:
But seems like it's not producing `.list` files? And might be preferable to make `.inc` files (cf. LLVM) instead, so you don't need extra Makefile rules, but can directly #include them.